### PR TITLE
Added vm cpu memory modal

### PR DESCRIPTION
**Pull request by metalice**
Signed-off-by: Matan Schatzman <mschatzm@redhat.com>

## 📝 Description

> Initially modal was created for VMI but it is currently not needed, so moved to VM, where it's needed, should be edited to match to VM.

## 🎥 Demo

> Please add a video or an image of the behavior/changes

### CODE REVIEW

File: src/utils/components/CPUMemoryModal/CpuMemoryModal.tsx [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+      className="cpu-memory-modal"
+      variant={ModalVariant.small}
+      onClose={() => setShowFlavorModal(false)}
+      actions={[
+        <Button
+          key="confirm"
+          variant={ButtonVariant.primary}
+          onClick={() => {
+            return null;
+            // k8sUpdate<V1VirtualMachineInstance>({
```
Comment (glekner) [inline src/utils/components/CPUMemoryModal/CpuMemoryModal.tsx:63]: this is why we introduced `immer` so we dont need to care about untouched parts.

delete this for now

File: src/utils/components/CPUMemoryModal/CpuMemoryModal.tsx [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+  TitleSizes,
+} from '@patternfly/react-core';
+
+import useTemplateDefaultCpuMemory from './hooks/useTemplateDefaultCpuMemory';
+import { getCPUCount, getMemorySize, memorySizesTypes } from './utils/CpuMemoryUtils';
+
+import './cpu-memory-modal.scss';
+
+type FlavorModalProps = {
+  vmi: V1VirtualMachineInstance;
```
Comment (glekner) [inline src/utils/components/CPUMemoryModal/CpuMemoryModal.tsx:27]: `obj: V1VirtualMachine | V1VirtualMachineInstance`

Comment (glekner) [inline src/utils/components/CPUMemoryModal/CpuMemoryModal.tsx:27]: `obj: k8sResourceCommon`

Comment (glekner) [reply src/utils/components/CPUMemoryModal/CpuMemoryModal.tsx:27]: also add `onSubmit: (flavor: Flavor) => Promise<void>`

Comment (glekner) [reply src/utils/components/CPUMemoryModal/CpuMemoryModal.tsx:27]: also add `onSubmit: (flavor: Flavor) => Promise<void>`

Comment (glekner) [reply src/utils/components/CPUMemoryModal/CpuMemoryModal.tsx:27]: also add `isOpen, onClose`

Comment (glekner) [reply src/utils/components/CPUMemoryModal/CpuMemoryModal.tsx:27]: also add `isOpen, onClose`

File: src/utils/components/CPUMemoryModal/CpuMemoryModal.tsx [hunk 3]
```diff
@@ -0,0 +1,172 @@
+import * as React from 'react';
+
+// import VirtualMachineInstanceModel from '@kubevirt-ui/kubevirt-api/console/models/VirtualMachineInstanceModel';
+import { V1VirtualMachineInstance } from '@kubevirt-ui/kubevirt-api/kubevirt';
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+// import { k8sUpdate } from '@openshift-console/dynamic-plugin-sdk';
```
Comment (glekner) [inline src/utils/components/CPUMemoryModal/CpuMemoryModal.tsx:6]: this modal shouldnt care, it should have a generic `onSubmit`

Comment (glekner) [inline src/utils/components/CPUMemoryModal/CpuMemoryModal.tsx:6]: this modal shouldnt care, it should have a generic `onSubmit`

File: src/utils/components/CPUMemoryModal/CpuMemoryModal.tsx [hunk 4]
```diff
@@ -0,0 +1,172 @@
+import * as React from 'react';
+
+// import VirtualMachineInstanceModel from '@kubevirt-ui/kubevirt-api/console/models/VirtualMachineInstanceModel';
```
Comment (glekner) [inline src/utils/components/CPUMemoryModal/CpuMemoryModal.tsx:3]: this modal shouldnt care, it should have a generic `onSubmit`

Comment (glekner) [inline src/utils/components/CPUMemoryModal/CpuMemoryModal.tsx:3]: this modal shouldnt care, it should have a generic `onSubmit`

File: src/utils/components/CPUMemoryModal/hooks/useTemplateDefaultCpuMemory.ts [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+  TemplateModel,
+  V1Template,
+  VirtualMachineModelRef,
+} from '@kubevirt-ui/kubevirt-api/console';
+import { V1VirtualMachine, V1VirtualMachineInstance } from '@kubevirt-ui/kubevirt-api/kubevirt';
+import { useK8sWatchResource } from '@openshift-console/dynamic-plugin-sdk';
+
+import { getCPUCount, getMemorySize } from '../utils/CpuMemoryUtils';
+
+type UseTemplateDefaultCpuMemory = (vmi: V1VirtualMachineInstance) => [
```
Comment (glekner) [inline src/utils/components/CPUMemoryModal/hooks/useTemplateDefaultCpuMemory.ts:12]: we should make this work with vm aswell, not just vmi

Comment (glekner) [inline src/utils/components/CPUMemoryModal/hooks/useTemplateDefaultCpuMemory.ts:12]: we should make this work with vm aswell, not just vmi

**Comment by avivtur** [comment]
can be closed
https://github.com/kubevirt-ui/kubevirt-plugin/pull/151

**Pull request closed without merging**